code/modules/mob/living/simple_animal/abnormality/teth/punishing_bird.dm

```diff
@@ -82,6 +82,14 @@
 	UnregisterSignal(SSdcs, COMSIG_GLOB_HUMAN_INSANE)
 	return ..()
 
+// Deletes perch if we moved while not breaching; Generally caused by cell swaps
+/mob/living/simple_animal/hostile/abnormality/punishing_bird/Move()
+	if(IsContained())
+		var/obj/structure/pbird_perch/P = locate() in get_turf(src)
+		QDEL_NULL(P)
+	. = ..()
+	PostSpawn()
```

Comment:
This would spawn a new perch in every spot it moves due to calling PostSpawn(), which has been overridden to spawn perch.

Reply:
That's the point - it deletes perch before moving and then spawns another on a new place.

Reply:
Saw the issue. Got it fixed.